src/session/src/lib.cairo

```diff
@@ -0,0 +1,148 @@
+use webauthn_session::signature::SignatureProofsTrait;
+use alexandria_data_structures::array_ext::ArrayTraitExt;
+use core::box::BoxTrait;
+use core::array::SpanTrait;
+use starknet::info::{TxInfo, get_tx_info, get_block_timestamp};
+use result::ResultTrait;
+use option::OptionTrait;
+use array::ArrayTrait;
+use core::{TryInto, Into};
+use starknet::{contract_address::ContractAddress};
+
+use webauthn_session::signature::{TxInfoSignature, FeltSpanTryIntoSignature, SignatureProofs};
+use webauthn_session::hash::{compute_session_hash, compute_call_hash};
+use alexandria_data_structures::merkle_tree::MerkleTreeImpl;
+
+
+use core::ecdsa::check_ecdsa_signature;
+
+mod hash;
+mod signature;
+
+#[cfg(test)]
+mod tests;
+
+type ValidationResult = Result<(), ()>;
+
+
+trait RevokedSessionKeyStorageTrait<T> {
+    fn contains_revoked_key(self: @T, key: felt252) -> bool;
+// fn add(self: T, key: felt252) -> Result<(), ()>;
```

Comment:
Is this needed?

Reply:
This is an interface analogous to:
https://github.com/argentlabs/starknet-plugin-account/blob/3c14770c3f7734ef208536d91bbd76af56dc2043/contracts/plugins/SessionKey.cairo#L44-L46
In the cairo0 implementation it's used here:
https://github.com/argentlabs/starknet-plugin-account/blob/3c14770c3f7734ef208536d91bbd76af56dc2043/contracts/plugins/SessionKey.cairo#L125-L129
This trait is in effect a layer of abstraction above some storage which keeps track of the keys that have been revoked.

Reply:
Ah I just mean the commented out part

Reply:
ah :))
the contract's storage provides a write method so this would be equivalent to the add method here. But since we don't use it internally in the validate function, you are right it's not needed (just forgot to delete the comment).